### PR TITLE
Add manual coverage checking for _process_auth()

### PR DESCRIPTION
**Pull request by jberglinds**

### CODE REVIEW

**Review by robban112 — DISMISSED** [review]

**Comment by ghost** [comment]
Needs multiple if-statements instead of and/or, will patch this.

**Comment by jojbak** [comment]
Don't merge the patch, needs some fixes.

**Comment by jberglinds** [comment]
The branch identifiers are way off. 
The code duplication makes this really hard to read, but if you think it's necessary, sure.

**Comment by jojbak** [comment]
@jberglinds Fixed, the code duplication is quite confusing yes but it works so I suggest we leave like that for now, we can always fix it when everything else is done.

File: httpie/input.py [hunk 1]
```diff
@@ -216,66 +216,81 @@ def _setup_standard_streams(self):
...
                 has_branched('process_auth', 4)
+            elif auth_type_set:
+                has_branched('process_auth', 5)
+
+            if not self.args.auth_type:
+                has_branched('process_auth', 6)
                 self.args.auth_type = default_auth_plugin.auth_type
             plugin = plugin_manager.get_auth_plugin(self.args.auth_type)()
 
             if plugin.auth_require and self.args.auth is None:
```
Comment (ghost) [inline httpie/input.py]: The "and self.args.auth is None" statement should be removed, since it has a separate if-statement below.

**Review by ghost — CHANGES_REQUESTED** [review]
Looks much cleaner now, just one minor thing

**Review by ghost — APPROVED** [review]

**Pull request merged**